process/coordinator/process_test.go

```diff
@@ -484,6 +497,50 @@ func TestTransactionCoordinator_CreateMarshalizedData(t *testing.T) {
 	mrBody, mrTxs := tc.CreateMarshalizedData(createTestBody())
 	assert.Equal(t, 0, len(mrTxs))
 	assert.Equal(t, 1, len(mrBody))
+	assert.Equal(t, len(createTestBody()), len(mrBody[1]))
+}
+
+func TestTransactionCoordinator_CreateMarshalizedDataWithTxsAndScr(t *testing.T) {
+	t.Parallel()
+
+	interimContainer := createInterimProcessorContainer()
+	tc, err := NewTransactionCoordinator(
+		mock.NewMultiShardsCoordinatorMock(5),
+		&mock.AccountsStub{},
+		mock.NewPoolsHolderFake(),
+		&mock.RequestHandlerMock{},
+		createPreProcessorContainer(),
+		interimContainer,
+	)
+	assert.Nil(t, err)
+	assert.NotNil(t, tc)
+
+	scrs := make([]data.TransactionHandler, 0)
+	body := block.Body{}
+	body = append(body, createMiniBlockWithOneTx(0, 1, block.TxBlock, []byte("tx_hash1")))
+
+	scr := &smartContractResult.SmartContractResult{SndAddr: []byte("snd"), RcvAddr: []byte("rcv"), Value: big.NewInt(99)}
+	scrHash, _ := core.CalculateHash(&mock.MarshalizerMock{}, &mock.HasherMock{}, scr)
+	scrs = append(scrs, scr)
+	body = append(body, createMiniBlockWithOneTx(0, 1, block.SmartContractResultBlock, scrHash))
+
+	scr = &smartContractResult.SmartContractResult{SndAddr: []byte("snd"), RcvAddr: []byte("rcv"), Value: big.NewInt(199)}
+	scrHash, _ = core.CalculateHash(&mock.MarshalizerMock{}, &mock.HasherMock{}, scr)
+	scrs = append(scrs, scr)
+	body = append(body, createMiniBlockWithOneTx(0, 1, block.SmartContractResultBlock, scrHash))
+
+	scr = &smartContractResult.SmartContractResult{SndAddr: []byte("snd"), RcvAddr: []byte("rcv"), Value: big.NewInt(299)}
+	scrHash, _ = core.CalculateHash(&mock.MarshalizerMock{}, &mock.HasherMock{}, scr)
+	scrs = append(scrs, scr)
+	body = append(body, createMiniBlockWithOneTx(0, 1, block.SmartContractResultBlock, scrHash))
+
+	scrInterimProc, _ := interimContainer.Get(block.SmartContractResultBlock)
+	_ = scrInterimProc.AddIntermediateTransactions(scrs)
+
+	mrBody, mrTxs := tc.CreateMarshalizedData(body)
+	assert.Equal(t, 1, len(mrTxs))
+	assert.Equal(t, len(scrs), len(mrTxs[1]))
```

Comment:
should also check unmarshaled mrTxs to match provided body

Reply:
done